src/adapters/aave-v2/products/pool/aaveV2BasePoolAdapter.ts

```diff
@@ -0,0 +1,361 @@
+import { ERC20 } from '../../../../core/utils/getTokenMetadata'
+import { logger } from '../../../../core/utils/logger'
+import { formatProtocolTokenArrayToMap } from '../../../../core/utils/protocolTokenToMap'
+import {
+  GetAprInput,
+  GetApyInput,
+  GetEventsInput,
+  GetPositionsInput,
+  GetPricesInput,
+  GetProfitsInput,
+  GetTotalValueLockedInput,
+  IProtocolAdapter,
+  MovementsByBlock,
+  ProfitsTokensWithRange,
+  ProtocolAprToken,
+  ProtocolApyToken,
+  ProtocolDetails,
+  ProtocolPricePerShareToken,
+  ProtocolToken,
+  ProtocolTotalValueLockedToken,
+  TokenType,
+} from '../../../../types/adapter'
+import { AaveV2PoolMetadata } from '../../buildMetadata'
+
+export abstract class AaveV2BasePoolAdapter implements IProtocolAdapter {
+  private metadata: AaveV2PoolMetadata
+  private provider: ethers.providers.StaticJsonRpcProvider
+  protected chainId: Chain
+
+  constructor({
+    metadata,
+    provider,
+    chainId,
+  }: {
+    metadata: AaveV2PoolMetadata
+    provider: ethers.providers.StaticJsonRpcProvider
+    chainId: Chain
+  }) {
+    this.metadata = metadata
+    this.provider = provider
+    this.chainId = chainId
+  }
+
+  abstract getProtocolDetails(): ProtocolDetails
+
+  async getProtocolTokens(): Promise<ERC20[]> {
+    return Object.values(this.metadata).map(
+      ({ protocolToken }) => protocolToken,
+    )
+  }
+
+  async getWithdrawals({
+    userAddress,
+    protocolTokenAddress,
+    fromBlock,
+    toBlock,
+  }: GetEventsInput): Promise<MovementsByBlock[]> {
+    const { protocolToken, underlyingToken } =
+      this.fetchProtocolTokenMetadata(protocolTokenAddress)
+
+    const protocolTokenContract =
+      this.protocolTokenContract(protocolTokenAddress)
+
+    const filter = protocolTokenContract.filters.Transfer(
+      userAddress,
+      ZERO_ADDRESS,
+    )
+
+    const eventResults = await protocolTokenContract.queryFilter<TransferEvent>(
+      filter,
+      fromBlock,
+      toBlock,
+    )
+
+    return await this.eventUtil({
+      protocolToken,
+      underlyingToken,
+      eventResults,
+    })
+  }
+
+  async getDeposits({
+    userAddress,
+    protocolTokenAddress,
+    fromBlock,
+    toBlock,
+  }: GetEventsInput): Promise<MovementsByBlock[]> {
+    const { protocolToken, underlyingToken } =
+      this.fetchProtocolTokenMetadata(protocolTokenAddress)
+
+    const protocolTokenContract =
+      this.protocolTokenContract(protocolTokenAddress)
+
+    const filter = protocolTokenContract.filters.Transfer(
+      ZERO_ADDRESS,
+      userAddress,
+    )
+
+    const eventResults = await protocolTokenContract.queryFilter<TransferEvent>(
+      filter,
+      fromBlock,
+      toBlock,
+    )
+
+    return await this.eventUtil({
+      protocolToken,
+      underlyingToken,
+      eventResults,
+    })
+  }
+
+  async getClaimedRewards(_input: GetEventsInput): Promise<MovementsByBlock[]> {
+    return []
+  }
+
+  async getOneDayProfit({
```

Comment:
As with `getWithdrawals` and `getDeposits`, this code is identical to the one being used in stargate, so we should extract it.